lib/Notification/FederationChatNotifier.php

```diff
@@ -0,0 +1,140 @@
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Talk\Notification;
+
+use OCA\Talk\Federation\Proxy\TalkV1\UserConverter;
+use OCA\Talk\Model\Attendee;
+use OCA\Talk\Model\Message;
+use OCA\Talk\Model\ProxyCacheMessage;
+use OCA\Talk\Participant;
+use OCA\Talk\Room;
+use OCP\AppFramework\Services\IAppConfig;
+use OCP\Notification\IManager;
+use OCP\Notification\INotification;
+
+class FederationChatNotifier {
+	public function __construct(
+		protected IAppConfig $appConfig,
+		protected IManager $notificationManager,
+		protected UserConverter $userConverter,
+	) {
+	}
+
+	/**
+	 * @param array{remoteServerUrl: string, sharedSecret: string, remoteToken: string, messageData: array{remoteMessageId: int, actorType: string, actorId: string, actorDisplayName: string, messageType: string, systemMessage: string, expirationDatetime: string, message: string, messageParameter: string, creationDatetime: string, metaData: string}, unreadInfo: array{unreadMessages: int, unreadMention: bool, unreadMentionDirect: bool}} $inboundNotification
+	 */
+	public function handleChatMessage(Room $room, Participant $participant, ProxyCacheMessage $message, array $inboundNotification): void {
+		$metaData = json_decode($inboundNotification['messageData']['metaData'] ?? '', true, flags: JSON_THROW_ON_ERROR);
+
+		if (isset($metaData[Message::METADATA_SILENT])) {
+			// Silent message, skip notification handling
+			return;
+		}
+
+		// Also notify default participants in one-to-one chats or when the admin default is "always"
+		$defaultLevel = $this->appConfig->getAppValueInt('default_group_notification', Participant::NOTIFY_MENTION);
+		if ($participant->getAttendee()->getNotificationLevel() === Participant::NOTIFY_MENTION
+			|| ($defaultLevel !== Participant::NOTIFY_NEVER && $participant->getAttendee()->getNotificationLevel() === Participant::NOTIFY_DEFAULT)) {
+			if ($this->isRepliedTo($room, $participant, $metaData)) {
+				$notification = $this->createNotification($room, $message, 'reply');
+				$notification->setUser($participant->getAttendee()->getActorId());
+				$this->notificationManager->notify($notification);
+			} elseif ($this->isMentioned($participant, $message)) {
+				$notification = $this->createNotification($room, $message, 'mention');
+				$notification->setUser($participant->getAttendee()->getActorId());
+				$this->notificationManager->notify($notification);
+			} elseif ($this->isMentionedAll($room, $message)) {
+				$notification = $this->createNotification($room, $message, 'mention_all');
+				$notification->setUser($participant->getAttendee()->getActorId());
+				$this->notificationManager->notify($notification);
+			}
+		} elseif ($participant->getAttendee()->getNotificationLevel() === Participant::NOTIFY_ALWAYS
```

Comment:
POST to `api/v4/room/{token}/notify` not implemented yet, returns 406

Reply:
Yeah, different todo